docs/flow/reference/components/discovery.kubernetes.md

```diff
@@ -0,0 +1,164 @@
+`own_namespace` | `bool`   | Include the namespace the agent is running in? | | no
+`names` | `[]string` | List of namespaces | | no
+
+### `selectors` block
+
+The `selectors` block contains optional label and field selectors to limit the discovery process to a subset of resources.
+See [https://kubernetes.io/docs/concepts/overview/working-with-objects/field-selectors/](https://kubernetes.io/docs/concepts/overview/working-with-objects/field-selectors/)
+and [https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/](https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/) to learn more about the possible
+filters that can be used.
+
+Note: When making decision about using field/label selector make sure that this
+is the best approach - it will prevent Prometheus from reusing single list/watch
+for all scrape configs. This might result in a bigger load on the Kubernetes API,
+because per each selector combination there will be additional LIST/WATCH. On the other hand,
+if you just want to monitor small subset of pods in large cluster it's recommended to use selectors.
+The decision, to use selectors or not depends on the particular situation.
+
+Name | Type | Description | Default | Required
+---- | ---- | ----------- | ------- | --------
+`role` | `string`   | role of the selector | | yes
+`label`| `string`   | label selector string | | no
+`field` | `string`   | field selector string | | no
+
+### `http_client_config` block
+
+The `http_client_config` block configures settings used to connect to the Kubernetes API server.
+Name | Type | Description | Default | Required
+---- | ---- | ----------- | ------- | --------
+`bearer_token`             | `secret`   | Use to set up the Bearer Token. | | no
+`bearer_token_file`        | `string`   | Use to set up the Bearer Token file. | | no
+`proxy_url`                | `string`   | Use to set up a proxy URL. | | no
+`follow_redirects`         | `bool`     | Whether the scraper should follow redirects. | `true` | no
+`enable_http_2`            | `bool`     | Whether the scraper should use HTTP2. | `true` | no
+
+The following sub-blocks are supported for `http_client_config`:
+
+Name | Description | Required
+---- | ----------- | --------
+[`basic_auth`](#basic_auth-block) | Configure basic_auth for authenticating against Kubernetes | no
+[`authorization`](#authorization-block) | Configure generic authorization against Kubernetes | no
+[`oauth2`](#oauth2-block) | Configure OAuth2 for authenticating against Kubernetes | no
+[`tls_config`](#tls_config-block) | Configure TLS settings for connecting to Kubernetes | no
+
+#### `basic_auth` block
+
+Name          | Type     | Description                                     | Default | Required
+------------- | -------- | ----------------------------------------------- | ------- | -------
+`username`      | `string`   | Setup of Basic HTTP authentication credentials. |         | no
```

Comment:
I think we should describe the fields rather than repeat "Setup of ... credentials."  Maybe just remove the Description column if there's nothing useful to say for all of these sub-blocks...